test/support/warning_filter.rb

```diff
@@ -0,0 +1,53 @@
+require 'pathname'
+require 'set'
+
+module RegexpParserTest
+  # Stream filter for warnings
+  class WarningFilter
```

Comment:
This class doesn't just observe. It extracts certain warnings from the stream and saves them.  `Warning::Extractor` seems better.